src/components/SignUp/Personality.js

```diff
@@ -45,10 +45,44 @@ class Personality extends Component {
     if (error) {
       alert(JSON.stringify(error));
     } else if (result.length !== 0) {
-      console.log(this.state.values);
+      const values = this.state.values.slice();
+
+      for (let i = 0, sliderIndex = 0; i < result.length; i += 2, sliderIndex++) {
+
+        // 0 means it has no direction
+        if (result[i].score === 0) {
+          continue;
+        } else {
+          const direction = this.findDirection(result[i].option);
+          values[sliderIndex] = result[i].score *= direction;
+          i--;
+        }
+      }
+
+      this.setState({ values: values });
     }
   }
 
+  /**
+   * Return -1 if given personality is from left side, or return 1
+   * if given personality is from right side.
+   * @param {string} option - given personality
+   * @returns {number} The direction. if nothing matches, return 0
+   */
+  findDirection(option) {
```

Comment:
Sure. First, the range of personality is basically -2 through 2 and 0 is indeed neutral value.
Second, option[0] goes left side and option[1] goes right side.
This function takes string of personality and find it's origin(from left side or right side) in order to figure out which way slider thumb should go.